components/tidb_query_datatype/src/codec/data_type/chunked_vec_common.rs

```diff
@@ -3,31 +3,33 @@
 #[macro_export]
 macro_rules! impl_chunked_vec_common {
     ($ty:ty) => {
-        pub fn from_slice(slice: &[Option<$ty>]) -> Self {
+        fn from_slice(slice: &[Option<$ty>]) -> Self {
             let mut x = Self::with_capacity(slice.len());
             for i in slice {
                 x.push(i.clone());
```

Comment:
For `ChunkedBytesVec` we could use `push_ref` to reduce overhead of `clone`. This may be done in future PRs.